PyMca5/PyMcaGui/io/TiledCatalogSelector.py

```diff
@@ -84,10 +84,38 @@ def url(self, value: str):
         
             Emits the 'url_changed' signal.
         """
+        old_value = self._url
         self._url = value
         self._url_buffer = value
-        # TODO: Perhaps this should only emit if the value is changed?
-        self.url_changed.emit()
+        if value != old_value:
+            self.url_changed.emit()
+
+    @property
+    def client(self):
+        return self._client
+
+    @client.setter
+    def client(self, _):
+        raise NotImplementedError("Call set_root_client() instead")
+
+    def set_root_client(self) -> None:
+        """Update the model with content from the new root client.
+        
+            Emits the 'table_changed' signal when a client is defined."""
+        try:
+            new_client = self.client_from_url(self.url)
+        except Exception as exception:
+            error_message = str(exception)
+            _logger.error(error_message)
+            self.client_connection_error.emit(error_message)
+            return
+
+        self._client = new_client
+        self.client_connected.emit(self._client.uri, str(self._client.context.api_uri))
+        self.node_path_parts = ()
+        self._current_page = 0
+        if self.client is not None:
+            self.table_changed.emit(self.node_path_parts)
```

Comment:
Adding some whitespace for logical separation.

```suggestion
        self._client = new_client
        self.client_connected.emit(self._client.uri, str(self._client.context.api_uri))

        self.node_path_parts = ()
        self._current_page = 0
        if self.client is not None:
            self.table_changed.emit(self.node_path_parts)
```